// discord-builder/dist/constants/questions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Answers = void 0;
const questions = [
    {
        type: "select",
        name: "language",
        choices: [
            { title: "typescript", value: "ts" },
            { title: "javascript", value: "js" }
        ],
        message: "please select a language"
    },
    {
        type: "text",
        name: "directory",
        initial: process.cwd(),
        message: "please enter a directory"
    },
    {
        type: "password",
        name: "token",
        message: "please enter a bot token"
    },
    {
        type: "text",
        name: "prefix",
        initial: "!",
        message: "please enter a prefix"
    },
    {
        type: "select",
        name: "loader",
        choices: [
            { title: "npm", value: "npm" },
            { title: "yarn", value: "yarn" }
        ],
        message: "please select a installer"
    }
];
let ques = ["language", "directory", "token", "loader"];
exports.Answers = Object.freeze({
    language: {
        required: true,
        values: ["ts", "js"]
    },
    directory: {
        required: true
    },
    token: {
        required: true
    },
    prefix: {
        required: true
    },
    loader: {
        required: true,
        values: ["npm", "yarn"]
    },
});
exports.default = {
    PromptQuestions: questions,
    ques
};
